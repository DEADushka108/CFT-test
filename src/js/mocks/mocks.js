export * from './const/const';
export * from './posts/posts';
export * from './store/store';
export * from './users/users';
export * from './comments/comments';
