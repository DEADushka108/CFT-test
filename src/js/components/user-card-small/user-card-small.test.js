import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import {store, user} from '../../mocks/mocks';
import UserCardSmall from './user-card-small';

it(`Should render UserSmallCard component correctly`, () => {
  const tree = renderer
    .create((
      <Provider store={store}>
        <MemoryRouter>
          <UserCardSmall
            userInfo={user}
          />
        </MemoryRouter>
      </Provider>
    ), {
      createNodeMock() {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
