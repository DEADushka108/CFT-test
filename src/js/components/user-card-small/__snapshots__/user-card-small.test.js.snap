// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render UserSmallCard component correctly 1`] = `
<article
  className="users-block__user-card-small user-card-small"
>
  <div
    className="user-card-small__image"
  >
    <img
      alt="Bret"
      className="user-card-small__user-image"
      src="./img/content/no-user.png"
    />
    <div>
      <p
        className="user-card-small__name"
      >
        Leanne Graham
      </p>
    </div>
  </div>
  <div
    className="user-card-small__user-info"
  >
    <ul
      className="user-card-small__info-list"
    >
      <li
        className="user-card-small__item"
      >
        <p
          className="user-card-small__text"
        >
          Username:
        </p>
        <a
          className="user-card-small__link"
          href="/user/1"
          onClick={[Function]}
        >
          Bret
        </a>
      </li>
      <li
        className="user-card-small__item"
      >
        <p
          className="user-card-small__text"
        >
          Email:
        </p>
        <a
          className="user-card-small__link"
          href="mailto:Sincere@april.biz"
        >
          Sincere@april.biz
        </a>
      </li>
      <li
        className="user-card-small__item"
      >
        <p
          className="user-card-small__text"
        >
          Phone:
        </p>
        <a
          className="user-card-small__link"
          href="tel:1-770-736-8031 x56442"
        >
          1-770-736-8031 x56442
        </a>
      </li>
      <li
        className="user-card-small__item"
      >
        <p
          className="user-card-small__text"
        >
          Website:
        </p>
        <a
          className="user-card-small__link"
          href="hildegard.org"
        >
          hildegard.org
        </a>
      </li>
    </ul>
  </div>
</article>
`;
